app/src/main/java/com/example/drawingApp/MainActivity.kt

```diff
@@ -2,101 +2,121 @@ package com.example.drawingApp
 
 import android.os.Bundle
 import android.os.StrictMode
-import android.view.LayoutInflater
-import android.view.View
-import android.widget.EditText
+import android.util.Log
 import android.widget.TextView
-import android.widget.Toast
 import androidx.activity.viewModels
 import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatActivity
-import androidx.core.view.isVisible
-import androidx.core.widget.addTextChangedListener
+
 
 class MainActivity : AppCompatActivity() {
     private val model: DrawingViewModel by viewModels()
+    private var isAlertDialogOpen = false
+    private var isLayerSheetOpen = false
+
+    var botSheetObj: DialogUtility.SheetObject? = null
+    var alertDialog: AlertDialog? = null
```

Comment:
If you have an object for both of these, you could probably get rid of your two booleans and just check if these are null or not

Reply:
removed booleans and implemented null checks instead